### ISSUE TITLE
ClientBuilder::setPrivateKey should have a variant where I can set the pem-content directly

### ISSUE THREAD

**Issue by hanswesterbeek** [okta/okta-sdk-java#486]
Right now, it allows me to set a String which contains a hard file-system-path. This is not appropriate in many cloud-native environments where configuration-properties like this are loaded from a service, such as AWS Secrets Manager.

Also, it is inconvenient for the pem file to _have to_ be in a physical file. In containerized cloud-native environments access to a filesystem is not as normal as it was in the old days.

It will also be much more flexible this way. Maybe passing an `InputStream` is even better.

So please allow the pem content to be set something like this:

```java

String pemContent = readPem();// so this I take care of myself
var client = Clients.builder()
                .setPrivatePem(pemContent) // pass the whole pem, avoid reading the FS
                .setOrgUrl("https://" + oktaDomain)  // e.g. https://dev-123456.okta.com
                .setAuthorizationMode(AuthorizationMode.PRIVATE_KEY)
                .setClientId("my-id")
                .setScopes(Set.of("okta.users.read"))
                .setCacheManager(Caches.newDisabledCacheManager())
                .build();
```

**Comment by sergiishamrai-okta** [comment]
@hanswesterbeek , Thanks for the suggestion, I've placed an item on our backlog to address this.
Since this is an open-source library, feel free to submit a pull request with your proposed changes for consideration.
Thanks for using Okta!

**Comment by hanswesterbeek** [comment]
If I provide that PR, how long until you merge and release? :)

**Comment by sergiishamrai-okta** [comment]
Unfortunately, we can't commit to a specific timeline since the work is placed into a queue for prioritization. But we can review your PR when it's ready and re-prioritize release if needed.

**Comment by hanswesterbeek** [comment]
It shouldn't be hard to add the feature. I guess I'll make it so that `setPrivateKeyContent` and `setPrivateKey` are mutually exclusive on the builder to avoid end-user confusion, and throwing a clear `IAE`.

So I forked and tried to build locally, as per instructions: https://github.com/okta/okta-sdk-java/wiki/Build-It

Unfortunately I get this jaxb issue:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Okta Java SDK 2.0.1-SNAPSHOT:
[INFO] 
[INFO] Okta Java SDK ...................................... FAILURE [  1.037 s]
[INFO] Okta Java SDK :: Swagger Templates ................. SKIPPED
[INFO] Okta Java SDK :: API ............................... SKIPPED
[INFO] Okta Java SDK :: Impl .............................. SKIPPED
[INFO] Okta Java SDK :: Extensions :: HTTP Client ......... SKIPPED
[INFO] Okta Java SDK :: Extensions :: OkHttp Client ....... SKIPPED
[INFO] Okta Java SDK :: Integration Tests ................. SKIPPED
[INFO] Okta Java SDK :: Examples .......................... SKIPPED
[INFO] Okta Java SDK :: Examples :: Quickstart Code ....... SKIPPED
[INFO] Okta Java SDK :: Coverage .......................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.103 s
[INFO] Finished at: 2020-11-20T10:30:31+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jasig.maven:maven-notice-plugin:1.1.0:generate (default) on project okta-sdk-root: Execution default of goal org.jasig.maven:maven-notice-plugin:1.1.0:generate failed: A required class was missing while executing org.jasig.maven:maven-notice-plugin:1.1.0:generate: javax/xml/bind/JAXBException
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.jasig.maven:maven-notice-plugin:1.1.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/hwesterbeek/.m2/repository/org/jasig/maven/maven-notice-plugin/1.1.0/maven-notice-plugin-1.1.0.jar
[ERROR] urls[1] = file:/Users/hwesterbeek/.m2/repository/com/okta/okta-parent-build-support/16/okta-parent-build-support-16.jar
[ERROR] urls[2] = file:/Users/hwesterbeek/.m2/repository/org/jvnet/jaxb2_commons/jaxb2-basics-runtime/1.11.1/jaxb2-basics-runtime-1.11.1.jar
[ERROR] urls[3] = file:/Users/hwesterbeek/.m2/repository/com/googlecode/java-diff-utils/diffutils/1.3.0/diffutils-1.3.0.jar
[ERROR] urls[4] = file:/Users/hwesterbeek/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar
[ERROR] urls[5] = file:/Users/hwesterbeek/.m2/repository/commons-io/commons-io/2.6/commons-io-2.6.jar
[ERROR] urls[6] = file:/Users/hwesterbeek/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar
[ERROR] urls[7] = file:/Users/hwesterbeek/.m2/repository/org/apache/maven/shared/maven-dependency-tree/2.2/maven-dependency-tree-2.2.jar
[ERROR] urls[8] = file:/Users/hwesterbeek/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
[ERROR] urls[9] = file:/Users/hwesterbeek/.m2/repository/org/eclipse/aether/aether-util/0.9.0.M2/aether-util-0.9.0.M2.jar
[ERROR] urls[10] = file:/Users/hwesterbeek/.m2/repository/org/codehaus/plexus/plexus-utils/1.1/plexus-utils-1.1.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[project>com.okta.sdk:okta-sdk-root:2.0.1-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : javax.xml.bind.JAXBException
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
```

Then, when i comment-out the offending plugin and run `mvn test`, I get another plugin error.

```
[ERROR] Failed to execute goal org.codehaus.gmavenplus:gmavenplus-plugin:1.8.0:generateTestStubs (default) on project okta-sdk-api: Execution default of goal org.codehaus.gmavenplus:gmavenplus-plugin:1.8.0:generateTestStubs failed: A required class was missing while executing org.codehaus.gmavenplus:gmavenplus-plugin:1.8.0:generateTestStubs: Could not initialize class org.codehaus.groovy.reflection.ReflectionCache
```

Is there some instruction on the environment I am missing?

**Comment by arvindkrishnakumar-okta** [comment]
@hanswesterbeek what does `mvn -version` print?

**Comment by arvindkrishnakumar-okta** [comment]
Fixed in `v3.0.0` which got released today.

**Comment by hanswesterbeek** [comment]
Nice!
